### PR TITLE
Config / Enchance Ledger error handling for corner cases

### PR DESCRIPTION
**Pull request by superKalo**
Tweaks a few things around the Ledger error handling:

- Add: When Ledger throws unexpected error, include a fallback note what user can do to resolve it (because the errors incoming from the device are very vague like: "invalid channel")
- Change: LedgerKeyIterator now uses the LedgerController (instead of directly calling Ledger SDK methods)
- Change: Account Adder error emit now contains the actual error message (instead of a generic one)

### CODE REVIEW

**Review by Ivshti — APPROVED** [review]

File: src/interfaces/keyIterator.ts [hunk 1]
```diff
@@ -7,6 +7,8 @@ export interface KeyIterator {
   subType?: 'seed' | 'private-key'
   /** The wallet native SDK instance, if any exists */
   walletSDK?: any
+  /** The wallet controller, for the hardware wallets only */
+  controller?: any
```
Comment (PetromirDev) [inline src/interfaces/keyIterator.ts:11]: Why any?

File: src/libs/ledger/ledger.ts [hunk 2]
```diff
@@ -27,5 +27,8 @@ export const normalizeLedgerMessage = (error?: string): string => {
     return 'Rejected by your Ledger device.'
   }
 
-  return error
+  // Indicates a custom timeout error, no need to normalize
+  if (error.includes('Could not connect to your Ledger device for an extended period')) return error
+
+  return `Could not connect to your Ledger device. Please close any other apps that may be accessing your Ledger device (including wallet apps on your computer and web apps). Ensure your Ledger is responsive. Unplug and plug it again. Device error: ${error}`
```
Comment (PetromirDev) [inline src/libs/ledger/ledger.ts:33]: What if the device error is very long?

**Review by PetromirDev — CHANGES_REQUESTED** [review]

File: src/interfaces/keyIterator.ts (hunk 1 above)
Comment (superKalo) [reply src/interfaces/keyIterator.ts]: You're connect, there is a type for this, fixed in https://github.com/AmbireTech/ambire-common/pull/944/commits/d343850ac2ccafd5a05aa3e6ddb6e5bcdc6b1c70

File: src/libs/ledger/ledger.ts (hunk 2 above)
Comment (superKalo) [reply src/libs/ledger/ledger.ts:33]: Based on what I've seen, the device errors are relatively short, not descriptive and vague like "Invalid channel" and "Ledger Device is busy". I didn't manage to reproduce a lot longer error, so let's take the risk :v:

**Pull request merged**